src/main/java/com/aws/greengrass/util/platforms/windows/WindowsPlatform.java

```diff
@@ -287,16 +287,27 @@ public UserPrincipal lookupUserByName(Path path, String name) throws IOException
         // When running as the SYSTEM user, the name from the "user.name" property is "<hostname>$".
         // This name cannot be looked up normally, but it is well known, so we can account for it easily.
         // https://docs.microsoft.com/en-us/troubleshoot/windows-server/identity/compatibility-user-accounts-end-dollar-sign
-        if (name.endsWith("$")) {
+        if (name.equalsIgnoreCase(getComputerName() + "$")) {
             name = LOCAL_SYSTEM_USERNAME;
         }
         return path.getFileSystem().getUserPrincipalLookupService().lookupPrincipalByName(name);
     }
 
+    private static String getComputerName() {
+        try {
+            return Kernel32Util.getComputerName();
+        } catch (Win32Exception ex) {
+            String env = System.getenv("COMPUTERNAME");
+            if (env != null) {
+                return env;
+            }
+        }
+        return "";
```

Comment:
Some details: 
1. In catch block, I think we are trying again with `System.getenv("COMPUTERNAME");` could we add a comment on when do we expect the exception to be thrown? Also should we log.warn it?
2. Could you also add a comment on why we default to empty String and possible implications?

Reply:
Updated